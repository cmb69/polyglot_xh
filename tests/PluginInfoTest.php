<?php

/**
 * Copyright 2021 Christoph M. Becker
 *
 * This file is part of Polyglot_XH.
 *
 * Polyglot_XH is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Polyglot_XH is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Polyglot_XH.  If not, see <http://www.gnu.org/licenses/>.
 */

namespace Polyglot;

use ApprovalTests\Approvals;
use PHPUnit\Framework\TestCase;
use Plib\HtmlView as View;
use Polyglot\Infra\SystemChecker;

class PluginInfoTest extends TestCase
{
    public function testRendersPluginInfo(): void
    {
        $systemChecker = $this->createStub(SystemChecker::class);
        $systemChecker->method("checkVersion")->willReturn(false);
        $systemChecker->method("checkWritability")->willReturn(false);
        $sut = new PluginInfo(
            "./plugins/polyglot/",
            XH_includeVar("./languages/en.php", "plugin_tx")["polyglot"],
            $systemChecker,
            new View("./views/", XH_includeVar("./languages/en.php", "plugin_tx")["polyglot"])
        );
        $response = $sut();
        Approvals::verifyHtml($response->output());
    }
}
