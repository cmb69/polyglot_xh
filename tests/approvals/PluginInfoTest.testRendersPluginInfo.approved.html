<!-- polyglot plugin info -->
<h1>Polyglot 1.0</h1>
<div class="polyglot_syscheck">
  <h2>System check</h2>
  <p class="xh_fail">PHP version ≥ 7.1.0: failure</p>
  <p class="xh_fail">CMSimple_XH version ≥ 1.7.0: failure</p>
  <p class="xh_warning">./plugins/polyglot/css/ is writable: warning</p>
  <p class="xh_warning">./plugins/polyglot/cache/ is writable: warning</p>
  <p class="xh_warning">./plugins/polyglot/config/ is writable: warning</p>
  <p class="xh_warning">./plugins/polyglot/languages/ is writable: warning</p>
</div>
